src/components/InventoryV2/view/BatchDetailsModalForNursery.tsx

```diff
@@ -0,0 +1,544 @@
+
+  useEffect(() => {
+    if (availableSpecies && selectedBatch?.species_id) {
+      setSelectedSpecies(
+        availableSpecies.find((singleSpecies) => singleSpecies.id.toString() === selectedBatch.species_id)
+      );
+    }
+  }, [availableSpecies, selectedBatch?.species_id]);
+
+  useEffect(() => {
+    if (availableAccessions && selectedBatch?.accession_id) {
+      setSelectedAccession(
+        availableAccessions.find((accession) => accession.id.toString() === selectedBatch.accession_id)
+      );
+    }
+  }, [availableAccessions, selectedBatch?.accession_id]);
+
+  useEffect(() => {
+    if (availableSubLocations && selectedBatch?.subLocations) {
+      setSelectedSubLocations(
+        availableSubLocations.filter((subLocation) =>
+          (selectedBatch.subLocations || []).find(
+            (batchSubLocation) => subLocation.id.toString() === batchSubLocation.subLocation_id
+          )
+        )
+      );
+    }
+  }, [availableSubLocations, selectedBatch?.subLocations]);
+
+  useEffect(() => {
+    if (timeZone !== tz.id) {
+      setTimeZone(tz.id);
+    }
+  }, [tz.id, timeZone]);
+
+  useEffect(() => {
+    setRecord((previousRecord: FormRecord): FormRecord => {
+      if (!previousRecord) {
+        return previousRecord;
+      }
+      return {
+        ...previousRecord,
+        addedDate: addedDateChanged ? previousRecord.addedDate : getTodaysDateFormatted(timeZone),
+      };
+    });
+  }, [timeZone, setRecord, addedDateChanged]);
+
+  useEffect(() => {
+    const newBatch: Partial<CreateBatchRequestPayload> = {
+      addedDate: getTodaysDateFormatted(),
+      facilityId: nurseryId,
+      speciesId: undefined,
+      germinatingQuantity: undefined,
+      notReadyQuantity: undefined,
+      readyQuantity: undefined,
+    };
+    const initBatch = () =>
+      selectedBatch
+        ? convertSelectedBatchToFormRecord(selectedBatch)
+        : {
+            ...newBatch,
+            id: -1,
+            batchNumber: '',
+            latestObservedTime: '',
+            version: 0,
+          };
+
+    setRecord(initBatch());
+  }, [selectedBatch, nurseryId, setRecord, selectedOrganization]);
+
+  const MANDATORY_FIELDS = [
+    'facilityId',
+    'germinatingQuantity',
+    'notReadyQuantity',
+    'readyQuantity',
+    'addedDate',
+  ] as const;
+  type MandatoryField = (typeof MANDATORY_FIELDS)[number];
+
+  const hasErrors = () => {
+    if (record) {
+      return MANDATORY_FIELDS.some((field: MandatoryField) => record[field] === '' || record[field] === undefined);
+    }
+    return true;
+  };
+
+  const saveBatch = async () => {
+    if (record) {
+      if (hasErrors()) {
+        setValidateFields(true);
+        return;
+      }
+
+      let response: (Response & BatchData) | (Response & BatchId) | undefined;
+      let responseQuantities: Partial<Response> = { requestSucceeded: true, error: undefined };
+
+      if (record.id === -1) {
+        if (record.accessionId && isBatch(record)) {
```

Comment:
If there is an accession ID, we need to create a transfer (which will automatically create the batch)